pandas/_libs/tslibs/conversion.pyx

```diff
@@ -392,6 +392,44 @@ cdef _TSObject convert_datetime_to_tsobject(datetime ts, object tz,
     return obj
 
 
+cdef _TSObject create_tsobject_tz_using_offset(int64_t value,
```

Comment:
maybe you didn't see my comment, this should be in tzconversion.pyx

Reply:
no, it shouldnt.  tzconversion funcs basically all take either int64_t or ndarray[int64_t].  That module is not "aware" of _TSObject.

Reply:
hmm  the issue is don’t we group the tz conversion functions there?
so logically it makes sense to put this there

Reply:
its a specific subset of tzconversion functions that _need_ to be separated in order to make non-circular cimports work.  keeping TSObject _out_ of tzconversion maintains a well-defined scopes for both conversion and tzconversion:  conversion is about converting between datetime-like objects (datetime, Timestamp, TSObject, npy_datetimestruct).  tzconversion is about converting int64 timestamps between timezones.